File: src/core/account/infrastructure/repository/InMemoryAccountRepository.ts
import {AccountRepository} from "../../domain/AccountRepository";
import {Account} from '../../domain/Account';
import {AccountId} from "../../domain/AccountId";

export class InMemoryAccountRepository implements AccountRepository {

    private _database = new Map<string, Account>();

    findById(accountId: AccountId): Promise<Account|undefined> {
        const account = this._database.get(accountId.value);

        return Promise.resolve(account || undefined);
    }

    store(account: Account): Promise<void> {
        this._database.set(account.id.value, account);
        return Promise.resolve();
    }
}

File: python_django/core/account/infrastructure/in_memory_account_repository.py
from core.account.domain.accont_id import AccountId
from core.account.domain.account import Account
from core.account.domain.account_repository import AccountRepository


class InMemoryAccountRepository(AccountRepository):

    def __init__(self) -> None:
        self._database = {}

    def find_by_id(self, account_id: AccountId) -> Account:
        return self._database.get(account_id.value, None)

    def store(self, account: Account) -> None:
        self._database[account.id.value] = account
